include/chain_invoke.h

```diff
@@ -113,8 +222,46 @@ using namespace msl;
              */
             template<typename Obj, size_t ... Idx>
             constexpr void invokeImpl(std::index_sequence<Idx...>, Fx const & aFx, Obj & obj) {
+                using compare_obj_t = std::remove_const_t<std::remove_pointer_t<Obj>>;
+                static_assert(std::is_same_v<std::remove_const_t<class_t>, compare_obj_t>, "must be one type");
+
                 (obj.*aFx)(conditionalAddressOf<std::tuple_element_t<Idx, qalified_t>>(std::get<Idx>(tuple))...);
             }
+            template<typename Obj, size_t ... Idx>
+            constexpr void invokeImpl(std::index_sequence<Idx...>, Fx const & aFx, Obj* obj) {
+                using compare_obj_t = std::remove_const_t<std::remove_pointer_t<Obj>>;
+                static_assert(std::is_same_v<std::remove_const_t<class_t>, compare_obj_t>, "must be one type");
+
+                (obj->*aFx)(conditionalAddressOf<std::tuple_element_t<Idx, qalified_t>>(std::get<Idx>(tuple))...);
```

Comment:
See not any check that obj is not nullptr.

Reply:
Ok. I will fix. But better exception as i think. If all ok exception will be cheap.